PostHog/PostHogSDK.swift

```diff
@@ -468,30 +468,44 @@ let maxRetryDelay = 30.0
 
         let isIdentified = storageManager.isIdentified()
 
-        if distinctId != oldDistinctId, !isIdentified {
-            // We keep the AnonymousId to be used by decide calls and identify to link the previousId
-            storageManager.setAnonymousId(oldDistinctId)
-            storageManager.setDistinctId(distinctId)
-
-            storageManager.setIdentified(true)
-
-            let properties = buildProperties(distinctId: distinctId, properties: [
-                "distinct_id": distinctId,
-                "$anon_distinct_id": oldDistinctId,
-            ], userProperties: sanitizeDicionary(userProperties), userPropertiesSetOnce: sanitizeDicionary(userPropertiesSetOnce))
-            let sanitizedProperties = sanitizeProperties(properties)
-
-            queue.add(PostHogEvent(
-                event: "$identify",
-                distinctId: distinctId,
-                properties: sanitizedProperties
-            ))
+        // if identified -> distinctId == oldDistinctId
+        // if !identified -> distinctId != oldDistinctId
+        if (distinctId == oldDistinctId) == isIdentified {
```

Comment:
I find this approach harder to read that we even need a comment, I read 3 times and I am still unsure if it does the right thing :D but your call here.